model.py

```diff
@@ -67,67 +70,146 @@ def stop(self):
         logger.info('joining thread {}', self.name)
         self._thread.join()
 
+    def next_batch(self):
```

Comment:
This module works very similarly to the `next_batch(batch_size)` implementation using TensorFlow examples.